File: replace_inputs.js
/* eslint-disable @typescript-eslint/no-require-imports */
const fs = require('fs');
const path = require('path');

function processFile(filepath) {
    let content = fs.readFileSync(filepath, 'utf8');

    if (!/<input\b/.test(content)) return;

    if (filepath.endsWith(path.join('components', 'ui', 'input.tsx'))) return;

    let newContent = content.replace(/<input\b/g, '<Input');
    newContent = newContent.replace(/<\/input>/g, '</Input>');

    if (!newContent.includes('import { Input }')) {
        const importRegex = /^import\s+.*?;?\s*$/gm;
        let match;
        let lastMatch;
        while ((match = importRegex.exec(newContent)) !== null) {
            lastMatch = match;
        }

        const importStatement = '\nimport { Input } from "@/components/ui/input";';
        if (lastMatch) {
            const insertPos = lastMatch.index + lastMatch[0].length;
            newContent = newContent.slice(0, insertPos) + importStatement + newContent.slice(insertPos);
        } else {
            newContent = importStatement.trim() + '\n' + newContent;
        }
    }

    fs.writeFileSync(filepath, newContent, 'utf8');
    console.log(`Updated ${filepath}`);
}

function walkSync(dir, callback) {
    const files = fs.readdirSync(dir);
    files.forEach((file) => {
        var filepath = path.join(dir, file);
        const stats = fs.statSync(filepath);
        if (stats.isDirectory()) {
            walkSync(filepath, callback);
        } else if (stats.isFile() && (file.endsWith('.tsx') || file.endsWith('.jsx'))) {
            callback(filepath);
        }
    });
}

walkSync(path.join(process.cwd(), 'src'), processFile);

File: replace_inputs.py
import os
import re

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # check if <input is in content
    if not re.search(r'<input\b', content):
        return

    # Don't replace inside the Input component itself!
    if filepath.endswith('components\\ui\\input.tsx') or filepath.endswith('components/ui/input.tsx'):
        return

    # Replace <input and </input>
    new_content = re.sub(r'<input\b', '<Input', content)
    new_content = re.sub(r'</input>', '</Input>', new_content)

    # Add import if not present
    if 'import { Input }' not in new_content:
        # find the last import
        import_match = list(re.finditer(r'^import\s+.*?;?\s*$', new_content, re.MULTILINE))
        if import_match:
            last_import = import_match[-1]
            insert_pos = last_import.end()
            new_content = new_content[:insert_pos] + '\nimport { Input } from "@/components/ui/input";' + new_content[insert_pos:]
        else:
            new_content = 'import { Input } from "@/components/ui/input";\n' + new_content

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(new_content)
    
    print(f"Updated {filepath}")

def main():
    src_dir = os.path.join(os.getcwd(), 'src')
    for root, _, files in os.walk(src_dir):
        for file in files:
            if file.endswith(('.tsx', '.jsx')):
                process_file(os.path.join(root, file))

if __name__ == "__main__":
    main()
